com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -586,5 +586,167 @@ internal NetworkBehaviour GetNetworkBehaviourAtOrderIndex(ushort index)
 
             return ChildNetworkBehaviours[index];
         }
+
+        /// <summary>
+        /// Used to serialize a NetworkObjects during scene syncrhonization that occurs
+        /// upon a client being approved or a scene transition.
+        /// </summary>
+        /// <param name="writer">writer into the outbound stream</param>
+        /// <param name="targetClientId">clientid we are targeting</param>
+        internal void SerializeSceneObject(PooledNetworkWriter writer, ulong targetClientId)
```

Comment:
```suggestion
        internal void SerializeSceneObject(NetworkWriter writer, ulong targetClientId)
```

we don't really have to rely on `PooledNetworkWriter`, we can just stick with `NetworkWriter`'s functionalities here (and `PooledNetworkWriter` will still be supported by that API design).